src/test/java/org/snomed/snowstorm/core/data/services/BranchMergeServiceTest.java

```diff
@@ -4245,6 +4245,271 @@ void testPromotionDoesNotResultInExtensionStealingAssociationMember() throws Ser
+
+		// 4. Create Extension project
+		String extProject = "MAIN/SNOMEDCT-TEST/projectA";
+		branchService.create(extProject);
+
+		// 5. Add translation to Concept
+		concept = conceptService.find(vehicleId, extProject);
+		concept.addDescription(new Description("Fordon").setTypeId(SYNONYM).setCaseSignificance(ci).setAcceptabilityMap(intAcceptable));
+		conceptService.update(concept, extProject);
+
+		// 6. Fix module error
+		concept = conceptService.find(vehicleId, intMain);
+		concept.setModuleId(CORE_MODULE);
+		concept = conceptService.update(concept, intMain);
+
+		// 7. Version International
+		codeSystem = codeSystemService.find("SNOMEDCT");
+		codeSystemService.createVersion(codeSystem, 20220228, "20220228");
+
+		// 8. Upgrade Extension
+		codeSystem = codeSystemService.find("SNOMEDCT-TEST");
+		codeSystemUpgradeService.upgrade(null, codeSystem, 20220228, false);
+
+		// 9. Rebase Extension project
+		review = getMergeReviewInCurrentState(extMain, extProject);
+		conflicts = reviewService.getMergeReviewConflictingConcepts(review.getId(), new ArrayList<>());
+		assertTrue(conflicts.size() == 1);
+
+		for (MergeReviewConceptVersions conflict : conflicts) {
+			Concept sourceConcept = conflict.getSourceConcept();
+			Concept autoMergedConcept = conflict.getAutoMergedConcept();
+			Concept targetConcept = conflict.getTargetConcept();
+
+			// Target is a version behind
+			assertEquals(20220228, sourceConcept.getEffectiveTimeI());
+			assertEquals(20220228, autoMergedConcept.getEffectiveTimeI());
+			assertEquals(20220131, targetConcept.getEffectiveTimeI());
+
+			// Target has previous modelling
+			assertEquals(CORE_MODULE, sourceConcept.getModuleId());
+			assertEquals(CORE_MODULE, autoMergedConcept.getModuleId());
+			assertEquals(MODEL_MODULE, targetConcept.getModuleId());
+
+			// Translation doesn't exist on Source
+			assertNull(getDescription(sourceConcept, "Fordon"));
+			assertNotNull(getDescription(autoMergedConcept, "Fordon"));
+			assertNotNull(getDescription(targetConcept, "Fordon"));
+
+			reviewService.persistManuallyMergedConcept(review, Long.parseLong(vehicleId), autoMergedConcept);
+		}
+
+		reviewService.applyMergeReview(review);
+
+		// 10. Final inspection
+		concept = conceptService.find(vehicleId, extProject);
+		assertEquals(20220228, concept.getEffectiveTimeI());
+		assertEquals(CORE_MODULE, concept.getModuleId());
+		assertEquals(3, concept.getDescriptions().size());
+	}
+
+	@Test
+	void testAutoMergeWhenRebasingVersionAhead() throws ServiceException, InterruptedException {
+		String intMain = "MAIN";
+		String extMain = "MAIN/SNOMEDCT-TEST";
+		Map<String, String> intPreferred = Map.of(US_EN_LANG_REFSET, descriptionAcceptabilityNames.get(PREFERRED), GB_EN_LANG_REFSET, descriptionAcceptabilityNames.get(PREFERRED));
+		Map<String, String> intAcceptable = Map.of(US_EN_LANG_REFSET, descriptionAcceptabilityNames.get(PREFERRED), GB_EN_LANG_REFSET, descriptionAcceptabilityNames.get(ACCEPTABLE));
```

Comment:
I think this is the test that fails on develop. 

Essentially: 

- lhs changes the concept's module (moves from module A to module B)
- rhs inactivates the concept (currently in module A)
- middle column ends up inactive in module A (because this is the module the concept is in on the task)

This is wrong because we're suggesting inactivation and also moving the module back to module A. With the new class, the middle column ends up inactive in module B.

Reply:
Good catch.